### DISCUSSION TITLE
Data unavailable for 2A-DPR on specific dates

### DISCUSSION THREAD

**Discussion by berberx** [ghiggi/gpm_api#71] [Q&A]
Hi,

I'm encountering an issue while using the gpm package to download data. The code I wrote based on the template you give is:
import gpm

username_pps = "berber"
password_pps = "ZayneandAmber_999"
base_dir = "/Users/amberrr/Desktop/rainfall_research"  # path to the directory where to download the data
gpm.define_configs(
    username_pps=username_pps, password_pps=password_pps, base_dir=base_dir
)

# You can check that the config file has been correctly created with:
configs = gpm.read_configs()
print(configs)

import datetime

product = "2A-DPR"
product_type = "RS"
version = 7

start_time = datetime.datetime(2023, 1, 1, 0, 1, 11)
end_time = datetime.datetime(2023, 12, 31, 0, 23, 5)

gpm.download(
    product=product,
    product_type=product_type,
    version=version,
    n_threads=2,
    start_time=start_time,
    end_time=end_time,
)

And when I try to run it in python, the following replies shows up:

The GPM-API config file has been updated successfully!
{'base_dir': '/Users/amberrr/Desktop/rainfall_research', 'username_pps': 'berber', 'password_pps': 'ZayneandAmber_999'}
No data found on PPS on date 2022-12-31 for product 2A-DPR (V07)
No data found on PPS on date 2023-01-01 for product 2A-DPR (V07)
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/gpm/io/download.py:996: GPMDownloadWarning: 'No data found on PPS on date 2023-01-01 for product 2A-DPR'
  status, available_version = _download_daily_data(
No data found on PPS on date 2023-01-02 for product 2A-DPR (V07)
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/gpm/io/download.py:996: GPMDownloadWarning: 'No data found on PPS on date 2023-01-02 for product 2A-DPR'
  status, available_version = _download_daily_data(
No data found on PPS on date 2023-01-03 for product 2A-DPR (V07)

It kept repeating for the rest of the dates I request.
Is this a known issue or an expected behavior? Are there any addition steps to fetch the data, or should I wait for these files to become available?

Thanks in advance for your help!

**Comment by ghiggi** [comment]
Hi Ruixi (@berberx), 

The username password you specified here above is for your EarthData Account, not the PPS ! 
Since you exposed your EarthData password publicly, please change your password. 

Read this page to understand how to register (1) a PPS and (2) a EarthData account: https://gpm-api.readthedocs.io/en/latest/03_quickstart.html

Once you registered to the PPS, you can check you have been granted access to PPS by login at: 
https://jsimpsonhttps.pps.eosdis.nasa.gov/text

**Comment by berberx** [comment]
Hi Gionata(@ghiggi ),

Thank you so much for your answer!
I have successfully registered to PPS yet this time. But when I try to run the code I found that I cannot reach to arthurhouftps.pps.eosdis.nasa.gov:

import gpm

configs = gpm.read_configs()
print(configs)

The following is the error:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/gpm/io/download.py", line 133, in check_valid_pps_credentials
    with ftplib.FTP_TLS(ftp_host) as ftp:
         ~~~~~~~~~~~~~~^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/ftplib.py", line 718, in __init__
    super().__init__(host, user, passwd, acct,
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
                     timeout, source_address, encoding=encoding)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/ftplib.py", line 121, in __init__
    self.connect(host)
    ~~~~~~~~~~~~^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/ftplib.py", line 158, in connect
    self.sock = socket.create_connection((self.host, self.port), self.timeout,
                ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                         source_address=self.source_address)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/socket.py", line 840, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/socket.py", line 975, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<python-input-13>", line 3, in <module>
    check_pps_ports_are_open()
    ~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/gpm/io/download.py", line 155, in check_pps_ports_are_open
    check_pps_available(verbose=False)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/gpm/io/download.py", line 147, in check_pps_available
    check_valid_pps_credentials(verbose=False)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/gpm/io/download.py", line 142, in check_valid_pps_credentials
    raise ValueError(f"The PPS server {ftp_host} is not available.")
ValueError: The PPS server arthurhouftps.pps.eosdis.nasa.gov is not available.

I think it might be some problems with the firewall or something? It will be helpful if you can teach me how to solve this!

Thank you so much!

**Comment by ghiggi** [comment]
It still seems a problem with the PPS credientials @berberx . Show me the output of 

```
configs = gpm.read_configs()
print(configs)
```
